packages/core/src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -95,14 +95,16 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
                 await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
             }
 
+            const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(`${gatewayId}`, { params: { method: methodId } }));
+
             if (isVaultedInstrument(paymentData)) {
                 await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
                 const { instrumentId } = paymentData;
+                const paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(payment.methodId);
 
-                return await this._executeWithVaulted(payment, instrumentId, shouldSetAsDefaultInstrument);
+                return await this._executeWithVaulted(payment, instrumentId, shouldSetAsDefaultInstrument, paymentMethod.clientToken);
```

Comment:
Do we need to check `paymentMethod.clientToken` is defined since `clientToken` is optional by the type definition?